### PR TITLE
PXT-345 Add .txt support for DocumentType

### PR DESCRIPTION
**Pull request by anushas**
This commit allows a .txt file to be added as a DocumentType and supports DocumentSplitter for it.

Testing: enhanced existing testcases and added a .txt specific testcase.

### CODE REVIEW

**Comment by aaron-siegel** [comment]
This change is [Reviewable](https://reviewable.io/reviews/pixeltable/pixeltable/389)

**Review by mkornacker — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pixeltable/pixeltable/389)* status: 0 of 6 files reviewed, 5 unresolved discussions (waiting on @anushas)

___
*[`pixeltable/utils/documents.py` line 94 at r2](https://reviewable.io/reviews/pixeltable/pixeltable/389#-OC_vULz4iqE352cRUJt:-OC_vUM-5MLIt_g1IY51:b-1ff09) ([raw file](https://github.com/pixeltable/pixeltable/blob/633fa9360aaf6801a4c072c0683e66d58cb5b195/pixeltable/utils/documents.py#L94)):*
> ```Python
>         return None
> 
> def get_txt_handle(path: str) -> Optional[str]:
> ```

Misleading naming: this is simply the text itself, not a handle to it.
___
*[`tests/test_document.py` line 245 at r2](https://reviewable.io/reviews/pixeltable/pixeltable/389#-OC_z0Oy1yp32Ok_EmyZ:-OC_z0Oy1yp32Ok_Emy_:b77q9z6) ([raw file](https://github.com/pixeltable/pixeltable/blob/633fa9360aaf6801a4c072c0683e66d58cb5b195/tests/test_document.py#L245)):*
> ```Python
>             pxt.drop_table('chunks')
> 
>     def test_doc_splitter_txt(self, reset_db) -> None:
> ```

There is already a test_doc_splitter, why special-case text docs?
___
*[`tests/test_document.py` line 258 at r2](https://reviewable.io/reviews/pixeltable/pixeltable/389#-OC_yynI9YiktQFOSj5x:-OC_yynI9YiktQFOSj5y:b-i6dwwm) ([raw file](https://github.com/pixeltable/pixeltable/blob/633fa9360aaf6801a4c072c0683e66d58cb5b195/tests/test_document.py#L258)):*
> ```Python
> 
>         chunks_t = pxt.create_view(
>                 'chunks', doc_t,
> ```

formatting
___
*[`pixeltable/iterators/document.py` line 157 at r2](https://reviewable.io/reviews/pixeltable/pixeltable/389#-OC_uo6SEnaHBtQd-SBv:-OC_uo6TAd-bzQe7FUA4:b-l6mwxj) ([raw file](https://github.com/pixeltable/pixeltable/blob/633fa9360aaf6801a4c072c0683e66d58cb5b195/pixeltable/iterators/document.py#L157)):*
> ```Python
>             assert self._doc_handle.txt_doc is not None
>             self._sections = self._txt_sections()
>             #self._sections = self._pdf_sections()
> ```

remove
___
*[`pixeltable/iterators/document.py` line 397 at r2](https://reviewable.io/reviews/pixeltable/pixeltable/389#-OC_w3WU7hW_0sq-LG0w:-OC_w3WU7hW_0sq-LG0x:b-bv3x1k) ([raw file](https://github.com/pixeltable/pixeltable/blob/633fa9360aaf6801a4c072c0683e66d58cb5b195/pixeltable/iterators/document.py#L397)):*
> ```Python
> 
>     def _txt_sections(self) -> Iterator[DocumentSection]:
>         """Create DocumentSections for text files."""
> ```

This could be more informative (you're creating a single section for the entire document).

Also add todo to add paragraphs.

**Review by anushas — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pixeltable/pixeltable/389)* status: 0 of 6 files reviewed, 5 unresolved discussions (waiting on @mkornacker)

___
*[`pixeltable/iterators/document.py` line 157 at r2](https://reviewable.io/reviews/pixeltable/pixeltable/389#-OC_uo6SEnaHBtQd-SBv:-OCe4ESB9qhDg2XsOToN:b-896fix) ([raw file](https://github.com/pixeltable/pixeltable/blob/633fa9360aaf6801a4c072c0683e66d58cb5b195/pixeltable/iterators/document.py#L157)):*
<details><summary><i>Previously, mkornacker (Marcel Kornacker) wrote…</i></summary><blockquote>

remove
</blockquote></details>

Done.
___
*[`pixeltable/iterators/document.py` line 397 at r2](https://reviewable.io/reviews/pixeltable/pixeltable/389#-OC_w3WU7hW_0sq-LG0w:-OCe4H7EELnkCXn7_BHy:b-896fix) ([raw file](https://github.com/pixeltable/pixeltable/blob/633fa9360aaf6801a4c072c0683e66d58cb5b195/pixeltable/iterators/document.py#L397)):*
<details><summary><i>Previously, mkornacker (Marcel Kornacker) wrote…</i></summary><blockquote>

This could be more informative (you're creating a single section for the entire document).

Also add todo to add paragraphs.
</blockquote></details>

Done.
___
*[`pixeltable/utils/documents.py` line 94 at r2](https://reviewable.io/reviews/pixeltable/pixeltable/389#-OC_vULz4iqE352cRUJt:-OCe4amUAuWB03dQfZ4x:b-896fix) ([raw file](https://github.com/pixeltable/pixeltable/blob/633fa9360aaf6801a4c072c0683e66d58cb5b195/pixeltable/utils/documents.py#L94)):*
<details><summary><i>Previously, mkornacker (Marcel Kornacker) wrote…</i></summary><blockquote>

Misleading naming: this is simply the text itself, not a handle to it.
</blockquote></details>

Done.
___
*[`tests/test_document.py` line 245 at r2](https://reviewable.io/reviews/pixeltable/pixeltable/389#-OC_z0Oy1yp32Ok_EmyZ:-OCe7PtH61TCsXTg4CXQ:b1ngz4g) ([raw file](https://github.com/pixeltable/pixeltable/blob/633fa9360aaf6801a4c072c0683e66d58cb5b195/tests/test_document.py#L245)):*
<details><summary><i>Previously, mkornacker (Marcel Kornacker) wrote…</i></summary><blockquote>

There is already a test_doc_splitter, why special-case text docs?
</blockquote></details>

as discussed, added comments.
___
*[`tests/test_document.py` line 258 at r2](https://reviewable.io/reviews/pixeltable/pixeltable/389#-OC_yynI9YiktQFOSj5x:-OCe9UC64i-mtRfjc6hn:b-896fix) ([raw file](https://github.com/pixeltable/pixeltable/blob/633fa9360aaf6801a4c072c0683e66d58cb5b195/tests/test_document.py#L258)):*
<details><summary><i>Previously, mkornacker (Marcel Kornacker) wrote…</i></summary><blockquote>

formatting
</blockquote></details>

Done.

File: tests/test_document.py [hunk 1]
```diff
@@ -230,8 +230,10 @@ def test_doc_splitter_headings(self, reset_db) -> None:
         for md_str in [','.join(t) for t in md_tuples]:
             print(f'{md_str=}')
             chunks_t = pxt.create_view(
-                'chunks', doc_t,
-                iterator=DocumentSplitter.create(document=doc_t.doc, separators='sentence', metadata=md_str))
+                            'chunks', doc_t,
```
Comment (mkornacker) [inline tests/test_document.py]: The formatting is still off. Did you run pylint?

See elsewhere in the code base for examples of multi-line function calls. Indentation is generally 4 spaces.

**Review by mkornacker — CHANGES_REQUESTED** [review]

File: tests/test_document.py (hunk 1 above)
Comment (anushas) [reply tests/test_document.py]: Hmm. I was keeping 4 spaces from where  the function call brace is at in previous line. But looks like we do 4 spaces from the beginning of newline. Adjusted. PTAL.

**Review by anushas — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pixeltable/pixeltable/389)* status: 0 of 6 files reviewed, 6 unresolved discussions (waiting on @mkornacker)

**Review by mkornacker — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pixeltable/pixeltable/389)* status: 0 of 6 files reviewed, 8 unresolved discussions (waiting on @anushas)

___
*[`tests/test_document.py` line 245 at r4](https://reviewable.io/reviews/pixeltable/pixeltable/389#-OCf36Mz8XJ3R2Qbw91h:-OCf36Mz8XJ3R2Qbw91i:b-5xdeyo) ([raw file](https://github.com/pixeltable/pixeltable/blob/0c6bb50e5739c9f9ec4ac948519599e310a3fc4a/tests/test_document.py#L245)):*
> ```Python
>             pxt.drop_table('chunks')
> 
>     # test_doc_splitter above already tests the behaviour
> ```

Function comments go in the docstring.
___
*[`tests/test_document.py` line 252 at r4](https://reviewable.io/reviews/pixeltable/pixeltable/389#-OCf3DSkAU1HwVTUZbuh:-OCf3DSlET61vWG1qj2i:b3kre) ([raw file](https://github.com/pixeltable/pixeltable/blob/0c6bb50e5739c9f9ec4ac948519599e310a3fc4a/tests/test_document.py#L252)):*
> ```Python
>         skip_test_if_not_installed('spacy')
> 
>         # DocumentSplitter does not support XML
> ```

?

**Review by anushas — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pixeltable/pixeltable/389)* status: 0 of 6 files reviewed, 8 unresolved discussions (waiting on @mkornacker)

___
*[`tests/test_document.py` line 245 at r4](https://reviewable.io/reviews/pixeltable/pixeltable/389#-OCf36Mz8XJ3R2Qbw91h:-OCfCjXb6rrFuXLU11A_:bxc0hzo) ([raw file](https://github.com/pixeltable/pixeltable/blob/0c6bb50e5739c9f9ec4ac948519599e310a3fc4a/tests/test_document.py#L245)):*
<details><summary><i>Previously, mkornacker (Marcel Kornacker) wrote…</i></summary><blockquote>

Function comments go in the docstring.
</blockquote></details>

done. sorry, thought that was only for external api doctoring. 
___
*[`tests/test_document.py` line 252 at r4](https://reviewable.io/reviews/pixeltable/pixeltable/389#-OCf3DSkAU1HwVTUZbuh:-OCfCUqA8Te40Uhq4TJE:buzzxp3) ([raw file](https://github.com/pixeltable/pixeltable/blob/0c6bb50e5739c9f9ec4ac948519599e310a3fc4a/tests/test_document.py#L252)):*
<details><summary><i>Previously, mkornacker (Marcel Kornacker) wrote…</i></summary><blockquote>

?
</blockquote></details>

sorry, copy pasted from the other test. removed.

File: tests/test_document.py [hunk 2]
```diff
@@ -242,14 +242,16 @@ def test_doc_splitter_headings(self, reset_db) -> None:
                         _ = res[md_element]
             pxt.drop_table('chunks')
 
-    # test_doc_splitter above already tests the behaviour
-    # common for all document types. This tests adds specific
-    # verification for .txt file with specific content.
     def test_doc_splitter_txt(self, reset_db) -> None:
+        ''' Test the DocumentSplitter with a .txt file
```
Comment (mkornacker) [inline tests/test_document.py]: Docstrings start and end with """ (not ''').

**Review by mkornacker — APPROVED** [review]

File: tests/test_document.py (hunk 2 above)
Comment (anushas) [reply tests/test_document.py]: wow. Corrected.

**Pull request merged**